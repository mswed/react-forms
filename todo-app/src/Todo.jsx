import React from 'react';

const Todo = ({ text, id, remove }) => {
  const handleRemove = () => {
    console.log('Removing', id);
    remove(id);
  };
  return (
    <div>
      {text}
      <button onClick={handleRemove}>X</button>
    </div>
  );
};

export default Todo;
