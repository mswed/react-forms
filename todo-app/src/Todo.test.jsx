import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import Todo from './Todo';

it('Renders without crashing', async () => {
  render(<Todo />);
});

it('Matches snapshot', async () => {
  const { asFragment } = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
});
