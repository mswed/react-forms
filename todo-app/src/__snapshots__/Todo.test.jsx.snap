// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
